### PR TITLE
Fixes to address performance and accuracy issues

### PR DESCRIPTION
**Pull request by saipj**
This PR address the following changes:
1.  performance issues
2. accuracy issues
3. documentation changes
4. code refactoring

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -77,6 +79,66 @@ model = ORTModule(model)
 
 To see torch-ort in action, see https://github.com/microsoft/onnxruntime-training-examples, which shows you how to train the most popular HuggingFace models.
 
+# Accelerate inference for PyTorch models with ONNX Runtime (Preview)
+
+ONNX Runtime for PyTorch accelerates PyTorch model inference using ONNX Runtime.
```
Comment (ynimmaga) [inline Readme.md:84]: Should we change this to something like "ONNX Runtime for PyTorch is now extended to accelerate PyTorch model inference using ONNX Runtime"? because ONNX Runtime for PyTorch also accelerates training already?

File: Readme.md [hunk 2]
```diff
@@ -77,6 +79,66 @@ model = ORTModule(model)
...
+- `pip install torch-ort-inference`
+
+## Verify your installation
+
+Once you have created your environment, using Python, execute the following steps to validate that your installation is correct.
+
+1. Download a inference script
+
+   - `wget https://raw.githubusercontent.com/pytorch/ort/main/torch_ort_inference/tests/bert_for_sequence_classification.py`
+
```
Comment (ynimmaga) [inline Readme.md:105]: Currently I don't see the "tests" directory yet. Will this be added later along with the samples?

File: torch_ort_inference/docs/install.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    `pip install wget pandas sklearn transformers`
+
+3. Run the inference script
+
+    `python ./ort/torch_ort_inference/tests/bert_for_sequence_classification.py`
+
+4. Expected warnings
+
+   There are some warnings that are expected.
```
Comment (ynimmaga) [inline torch_ort_inference/docs/install.md:33]: Should we also add "Build from source" section here and describe how to build ONNX Runtime from source with the relevant execution provider? We should definitely add this in ONNX Runtime docs, but just wondering if we should add here too?

File: torch_ort_inference/pypi-readme.md [hunk 4]
```diff
@@ -1,34 +1,16 @@
-The torch-ort packages uses the PyTorch APIs to accelerate PyTorch models using ONNX Runtime.
+The torch-ort-inference packages uses the PyTorch APIs to accelerate PyTorch models using ONNX Runtime OpenVINO EP.
```
Comment (ynimmaga) [inline torch_ort_inference/pypi-readme.md]: Should this be just a single "package" instead of "package(s)"

File: torch_ort_inference/requirements.txt [hunk 5]
```diff
@@ -3,5 +3,5 @@
 # In an environment where a requirement is already met, the package will not be updated.
 # If a package of a specific version is needed, the package can be installed
 # with version being specified (see Readme.md).
-#onnxruntime-ortinference-openvino>=1.12.0
+onnxruntime-openvino>=1.11.0
 torch>=1.8
```
Comment (ynimmaga) [inline torch_ort_inference/requirements.txt:7]: Since we have not validated versions <1.11 and the versions <1.12 may not contain Aten fallback, it is better if we make 1.12 as the minimum version.

File: torch_ort_inference/setup.py [hunk 6]
```diff
@@ -20,18 +20,18 @@ def fetch_requirements(path):
...
 
 install_requires = fetch_requirements('requirements.txt')
 
-packages=[*setuptools.find_packages(), 'torch_ort_inference.configure']
+packages=[*setuptools.find_packages(), 'torch_ort.configure']
 
 setuptools.setup(
     name="torch_ort_inference",
     version=version_str,
     author="torch-ort contributors",
```
Comment (ynimmaga) [inline torch_ort_inference/setup.py:33]: Should this be "Intel"? Need to check with OV EP team and maintain consistency if we are publishing this recipe and package

File: torch_ort_inference/torch_ort/ortinferencemodule/__init__.py [hunk 7]
```diff
@@ -33,7 +33,6 @@ def _defined_from_envvar(name, default_value, warn=True):
 # NOTE: To *change* values in runtime, import onnxruntime.training.ortmodule and
 # assign them new values. Importing them directly do not propagate changes.
 ################################################################################
-ONNX_OPSET_VERSION = 14
 MINIMUM_RUNTIME_PYTORCH_VERSION_STR = "1.8.1"
```
Comment (ynimmaga) [inline torch_ort_inference/torch_ort/ortinferencemodule/__init__.py]: Should we limit the Minimum PyTorch version to 1.12?

File: torch_ort_inference/torch_ort/ortinferencemodule/__init__.py [hunk 8]
```diff
@@ -33,7 +33,6 @@ def _defined_from_envvar(name, default_value, warn=True):
 # NOTE: To *change* values in runtime, import onnxruntime.training.ortmodule and
 # assign them new values. Importing them directly do not propagate changes.
 ################################################################################
-ONNX_OPSET_VERSION = 14
```
Comment (ynimmaga) [inline torch_ort_inference/torch_ort/ortinferencemodule/__init__.py:36]: Does the same opset version from ORTModule also work for ORTInferenceModule?

File: Readme.md (hunk 2 above)
Comment (saipj) [reply Readme.md:105]: That's right.

File: torch_ort_inference/docs/install.md (hunk 3 above)
Comment (saipj) [reply torch_ort_inference/docs/install.md:33]: Since torch/ort is an abstract layer and onnxruntime-openvino is added as dependency, would be good to maintain the parity with the existing torch-ort.

File: torch_ort_inference/pypi-readme.md (hunk 4 above)
Comment (saipj) [reply torch_ort_inference/pypi-readme.md]: Noted. Since we have only 1 package for torch-ort-inference unlike torch-ort.

File: torch_ort_inference/requirements.txt (hunk 5 above)
Comment (saipj) [reply torch_ort_inference/requirements.txt:7]: Noted. PR updated.

File: torch_ort_inference/torch_ort/ortinferencemodule/__init__.py (hunk 7 above)
Comment (saipj) [reply torch_ort_inference/torch_ort/ortinferencemodule/__init__.py]: Noted. Updated PR.

File: torch_ort_inference/torch_ort/ortinferencemodule/__init__.py (hunk 8 above)
Comment (saipj) [reply torch_ort_inference/torch_ort/ortinferencemodule/__init__.py:36]: MSFT mentioned that the latest opset_version used by torch.onnx during export should be used since it is for inference.

File: torch_ort_inference/setup.py (hunk 6 above)
Comment (saipj) [reply torch_ort_inference/setup.py:33]: This should be mainly by contributors to this repo. Hosting our package on PyPI would be from Intel domain.

File: torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py [hunk 9]
```diff
@@ -105,39 +97,35 @@ def forward_call(self, *inputs, **kwargs):
...
+        run_options = C.RunOptions()
+
+        # Pre-process inputs to make them compatible with onnxruntime
+        
+        # Use IO binding
+        io_binding = self._inference_session.io_binding()
+        _utils._create_iobinding(io_binding, inputs, self._onnx_models.exported_model, self._device)
+        
+        # Run inference session
+        self._inference_session.run_with_iobinding(io_binding, run_options)
```
Comment (ynimmaga) [inline torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py:116]: Do we need a session with io_binding? Does this provide any performance/latency benefits?

File: Readme.md (hunk 1 above)
Comment (saipj) [reply Readme.md:84]: Looks very long. Maybe we can keep the existing name and see if MSFT is good with it.

File: torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py (hunk 9 above)
Comment (saipj) [reply torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py:116]: we would still need io_binding since the inputs are converted from torch tensors to ortvalues.

**Review by ynimmaga — APPROVED** [review]
Thanks for the responses. LGTM.
